packages/core/src/Stores/client-store.js

```diff
@@ -2767,21 +2767,30 @@ export default class ClientStore extends BaseStore {
         const matchingSubscription = this.subscriptions?.[name]?.[key]?.sub;
         if (matchingSubscription) return { key, subscription: matchingSubscription };
 
-        const subscription = WS?.subscribe({
-            [name]: 1,
-            subscribe: 1,
-            ...(payload ?? {}),
-        });
+        try {
+            const subscription = await WS?.subscribe({
+                [name]: 1,
+                subscribe: 1,
+                ...(payload ?? {}),
+            });
 
-        this.addSubscription(name, key, subscription);
-        return { name, key, subscription };
+            this.addSubscription(name, key, subscription);
+            return { key, subscription };
+        } catch (e) {
+            // error handling needed
+            return {};
+        }
     };
 
     unsubscribeByKey = async (name, key) => {
         const matchingSubscription = this.subscriptions?.[name]?.[key]?.sub;
         if (matchingSubscription) {
-            await WS?.forget(this.subscriptions?.[name]?.[key]?.id);
-            delete this.subscriptions?.[name]?.[key];
+            try {
+                await WS?.forget(this.subscriptions?.[name]?.[key]?.id);
+                delete this.subscriptions?.[name]?.[key];
+            } catch (e) {
+                // error handling needed
```

Comment:
What error handling are we going to do?

Reply:
can we return this error so consumers can handle it?

Reply:
@farrah-deriv where do I need to return it? Create a field in the store and put the error there? In this case when I need to reset this error? Or do I need to create error object with fields for all currencies? Before my changes we just put error in useState in useSubscription hook and just return this error the the hook. But we didn't show/handle it anywhere.